### ISSUE TITLE
The wrong bounding box causes Chrome to refuse to render

### ISSUE THREAD

**Issue by yisibl** [fontello/svg2ttf#113]
Here is a simple example to help us improve the bounding box of the generated font. 

- Demo: https://at.alicdn.com/t/project/2575312/a0bdfc53-838a-4de8-b7cd-2196167f06da.html
- iconfont bug: https://github.com/thx/iconfont-plus/issues/2009

SVG file: 

```svg
<svg viewBox="0 0 62464 1024" version="1.1" xmlns="http://www.w3.org/2000/svg" width="200" height="200"><path d="M62464 0H0v1024h62464V0z" fill="#FF0000"></path></svg>
```

[OpenType Sanitizer](https://github.com/khaledhosny/ots) error prompt:

```shell
ERROR: head: Bad x dimension in the font bounding box (0, -3072)
ERROR: head: Failed to parse table
Failed to sanitize file!
```

Font Table

```xml
<head>
    <!-- Most of this table will be recalculated by the compiler -->
    <tableVersion value="1.0"/>
    <fontRevision value="1.0"/>
    <checkSumAdjustment value="0xd63f6d33"/>
    <magicNumber value="0x5f0f3cf5"/>
    <flags value="00000000 00001011"/>
    <unitsPerEm value="1024"/>
    <created value="Mon Jun 21 11:04:19 2021"/>
    <modified value="Mon Jun 21 11:04:19 2021"/>
    <xMin value="0"/>
    <yMin value="-128"/>
    <xMax value="-3072"/>
    <yMax value="896"/>
    <macStyle value="00000000 00000000"/>
    <lowestRecPPEM value="8"/>
    <fontDirectionHint value="2"/>
    <indexToLocFormat value="0"/>
    <glyphDataFormat value="0"/>
  </head>

  <hhea>
    <tableVersion value="0x00010000"/>
    <ascent value="896"/>
    <descent value="-128"/>
    <lineGap value="0"/>
    <advanceWidthMax value="62464"/>
    <minLeftSideBearing value="0"/>
    <minRightSideBearing value="0"/>
    <xMaxExtent value="-3072"/>
    <caretSlopeRise value="1"/>
    <caretSlopeRun value="0"/>
    <caretOffset value="0"/>
    <reserved0 value="0"/>
    <reserved1 value="0"/>
    <reserved2 value="0"/>
    <reserved3 value="0"/>
    <metricDataFormat value="0"/>
    <numberOfHMetrics value="2"/>
  </hhea>

  <maxp>
    <!-- Most of this table will be recalculated by the compiler -->
    <tableVersion value="0x10000"/>
    <numGlyphs value="2"/>
    <maxPoints value="4"/>
    <maxContours value="1"/>
    <maxCompositePoints value="0"/>
    <maxCompositeContours value="0"/>
    <maxZones value="2"/>
    <maxTwilightPoints value="0"/>
    <maxStorage value="10"/>
    <maxFunctionDefs value="10"/>
    <maxInstructionDefs value="0"/>
    <maxStackElements value="255"/>
    <maxSizeOfInstructions value="0"/>
    <maxComponentElements value="0"/>
    <maxComponentDepth value="0"/>
  </maxp>

  <OS_2>
    <!-- The fields 'usFirstCharIndex' and 'usLastCharIndex'
         will be recalculated by the compiler -->
    <version value="4"/>
    <xAvgCharWidth value="31744"/>
    <usWeightClass value="500"/>
    <usWidthClass value="5"/>
    <fsType value="00000000 00000000"/>
    <ySubscriptXSize value="649"/>
    <ySubscriptYSize value="716"/>
    <ySubscriptXOffset value="0"/>
    <ySubscriptYOffset value="143"/>
    <ySuperscriptXSize value="649"/>
    <ySuperscriptYSize value="716"/>
    <ySuperscriptXOffset value="0"/>
    <ySuperscriptYOffset value="491"/>
    <yStrikeoutSize value="50"/>
    <yStrikeoutPosition value="264"/>
    <sFamilyClass value="0"/>
    <panose>
      <bFamilyType value="2"/>
      <bSerifStyle value="0"/>
      <bWeight value="5"/>
      <bProportion value="3"/>
      <bContrast value="0"/>
      <bStrokeVariation value="0"/>
      <bArmStyle value="0"/>
      <bLetterForm value="0"/>
      <bMidline value="0"/>
      <bXHeight value="0"/>
    </panose>
    <ulUnicodeRange1 value="00000000 00000000 00000000 00000000"/>
    <ulUnicodeRange2 value="00000000 00000000 00000000 00000000"/>
    <ulUnicodeRange3 value="00000000 00000000 00000000 00000000"/>
    <ulUnicodeRange4 value="00000000 00000000 00000000 00000000"/>
    <achVendID value="PfEd"/>
    <fsSelection value="00000000 11000000"/>
    <usFirstCharIndex value="59053"/>
    <usLastCharIndex value="59053"/>
    <sTypoAscender value="896"/>
    <sTypoDescender value="-128"/>
    <sTypoLineGap value="92"/>
    <usWinAscent value="988"/>
    <usWinDescent value="128"/>
    <ulCodePageRange1 value="00000000 00000000 00000000 00000001"/>
    <ulCodePageRange2 value="00000000 00000000 00000000 00000000"/>
    <sxHeight value="0"/>
    <sCapHeight value="0"/>
    <usDefaultChar value="0"/>
    <usBreakChar value="0"/>
    <usMaxContext value="2"/>
  </OS_2>
```

**Comment by puzrin** [comment]
AFAIK term "wrong" is not precise:

- Bbox is calculated from real data and can not be wrong (after last fixes).
- BUT (!) too high values are "highly likely" caused by bad input.

Questions:

- We need concrete criteria of issue resolve
  - Probably "throw error on too big bbox values"
-  We need concrete bounds for bbox
  - Probably, use ones from Chrome/FF (need to extract somehow)

**Comment by yisibl** [comment]
Chrome and Firefox use OpenType Sanitizer to detect fonts, the code is here:
https://github.com/khaledhosny/ots/blob/ff09cd0b6b0dd210a4c1bafd61fcbbba21738751/src/head.cc#L66-L73

```c++
if (this->xmin > this->xmax) {
    return Error("Bad x dimension in the font bounding box (%d, %d)",
                  this->xmin, this->xmax);
  }
  if (this->ymin > this->ymax) {
    return Error("Bad y dimension in the font bounding box (%d, %d)",
                  this->ymin, this->ymax);
  }
```

**Comment by puzrin** [comment]
At first glance, those checks are not actual for svg2ttf. I mean, load errors happened due different checks.

**Comment by khaledhosny** [comment]
In the txt dump above xmin is 0 and xmax is negative, so xmin is greater than xmax.

**Comment by puzrin** [comment]
Hm... strange, i will check. Thanks for clarification.

**Comment by yisibl** [comment]
Yes, it is precisely because the font generated by svg2ttf has `xMin> xMax` that it will trigger OpenType Sanitizer to report an error.

```xml
 <xMin value="0"/>
 <xMax value="-3072"/>
```

**Comment by rlidwka** [comment]
The very first instruction in the svg path is `M62464`, which is exactly what xMax is calculated to be.

The field type allocated to it is: signed int16. We write 62464 there, which is the same as -3072 (62464+3072=65536).

So we have two issues here:
 - invalid glyph (or at least a glyph we can't safely convert to ttf)
 - no boundary checks

What do you expect svg2ttf to do here? Would an error (if glyph boundaries fall outside of ±32767) be enough?

**Comment by rlidwka** [comment]
Fixed, now this font should output an error.

**Comment by yisibl** [comment]
Looks good, thanks!

**Comment by puzrin** [comment]
Published 6.0.2
